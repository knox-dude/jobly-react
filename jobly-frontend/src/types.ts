export interface Company {
  handle: string;
  name: string;
  description: string;
  numEmployees?: number;
  logoUrl?: string;
  jobs?: Job[];
}

export interface Job {
  id: string;
  title: string;
  salary?: number;
  equity?: number;
  companyHandle?: string;
  companyName?: string;
  company?: Company;
}

export interface User {
  username: string;
  firstName: string;
  lastName: string;
  email: string;
  isAdmin: boolean;
  applications: string[];
}
