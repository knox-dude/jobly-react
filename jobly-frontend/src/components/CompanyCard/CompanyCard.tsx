import { Card, CardBody, CardTitle, CardSubtitle, CardText } from "reactstrap"
import { Company } from "@/types";
import { useNavigate } from "react-router-dom";


function CompanyCard({handle, name, description, numEmployees, logoUrl}: Company) {

  const navigate = useNavigate();

  return (
    <Card onClick={() => navigate(`/companies/${handle}`)} style={{cursor:"pointer"}} className="company-card flex-row m-2">
      <CardBody className="col-10">
        <div className="h-100 d-flex flex-column justify-content-between">
          <div>
            <CardTitle tag="h5">{name}</CardTitle>
            <CardSubtitle className="mb-2 text-muted" tag="h6">{description}</CardSubtitle>
          </div>
          <CardText>Employees: {numEmployees}</CardText>
        </div>

      </CardBody>

      <div className="container col-2 p-2">
        <img className="img-fluid" src={logoUrl} alt={name} />
      </div>
    </Card>
  )
}


export default CompanyCard;
