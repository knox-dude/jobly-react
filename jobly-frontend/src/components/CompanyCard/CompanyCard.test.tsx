import { render, screen } from "@testing-library/react";
import CompanyCard from "@/components/CompanyCard/CompanyCard";
import { Company } from "@/types";
import { BrowserRouter } from "react-router-dom";

const mockCompany: Company = {
  handle: "test-handle",
  name: "Test Company",
  description: "This is a test company.",
  numEmployees: 100,
  logoUrl: "https://example.com/logo.png",
};

describe("CompanyCard", () => {

  it("passes the snapshot test", () => {
    const {asFragment} = render(
      <BrowserRouter>
        <CompanyCard {...mockCompany} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders and displays the company name", () => {
    render(
      <BrowserRouter>
        <CompanyCard {...mockCompany} />
      </BrowserRouter>
    );
    const companyName = screen.getByText(mockCompany.name);
    expect(companyName).toBeInTheDocument();
  });

  it("renders the company description", () => {
    render(
      <BrowserRouter>
        <CompanyCard {...mockCompany} />
      </BrowserRouter>
    );
    const companyDescription = screen.getByText(mockCompany.description);
    expect(companyDescription).toBeInTheDocument();
  });

  it("renders the number of employees", () => {
    render(
      <BrowserRouter>
        <CompanyCard {...mockCompany} />
      </BrowserRouter>
    );
    const employeesCount = screen.getByText(
      `Employees: ${mockCompany.numEmployees}`
    );
    expect(employeesCount).toBeInTheDocument();
  });
});
