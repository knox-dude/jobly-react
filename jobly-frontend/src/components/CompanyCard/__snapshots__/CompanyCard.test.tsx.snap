// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyCard passes the snapshot test 1`] = `
<DocumentFragment>
  <div
    class="company-card flex-row m-2 card"
    style="cursor: pointer;"
  >
    <div
      class="col-10 card-body"
    >
      <div
        class="h-100 d-flex flex-column justify-content-between"
      >
        <div>
          <h5
            class="card-title"
          >
            Test Company
          </h5>
          <h6
            class="mb-2 text-muted card-subtitle"
          >
            This is a test company.
          </h6>
        </div>
        <p
          class="card-text"
        >
          Employees: 100
        </p>
      </div>
    </div>
    <div
      class="container col-2 p-2"
    >
      <img
        alt="Test Company"
        class="img-fluid"
        src="https://example.com/logo.png"
      />
    </div>
  </div>
</DocumentFragment>
`;
